spec/lib/preservation_catalog/s3_spec.rb

```diff
@@ -21,26 +21,81 @@
   end
 
   describe 'config' do
-    let(:config) { described_class.client.config }
-    let(:trio) do
-      {
-        'AWS_SECRET_ACCESS_KEY' => 'secret',
-        'AWS_ACCESS_KEY_ID' => 'some_key',
-        'AWS_REGION' => 'us-east-1'
-      }
-    end
+    context 'with access key and region env vars' do
+      let(:config) { described_class.client.config }
+      let(:trio) do
+        {
+          'AWS_SECRET_ACCESS_KEY' => 'secret',
+          'AWS_ACCESS_KEY_ID' => 'some_key',
+          'AWS_REGION' => 'us-east-1'
+        }
+      end
 
-    around do |example|
-      old_vals = trio.keys.zip(ENV.values_at(*trio.keys)).to_h
-      trio.each { |k, v| ENV[k] = v }
-      example.run
-      old_vals.each { |k, v| ENV[k] = v }
+      around do |example|
+        old_vals = trio.keys.zip(ENV.values_at(*trio.keys)).to_h
+        trio.each { |k, v| ENV[k] = v }
+        example.run
+        old_vals.each { |k, v| ENV[k] = v }
+      end
+      it 'pulls from ENV vars' do
+        expect(config.region).to eq 'us-east-1'
+        expect(config.credentials).to be_an(Aws::Credentials)
+        expect(config.credentials).to be_set
+        expect(config.credentials.access_key_id).to eq 'some_key'
+      end
     end
-    it 'pulls from ENV vars' do
-      expect(config.region).to eq 'us-east-1'
-      expect(config.credentials).to be_an(Aws::Credentials)
-      expect(config.credentials).to be_set
-      expect(config.credentials.access_key_id).to eq 'some_key'
+
+    context 'pointing the client to shared credentials' do
+      let(:config) { described_class.client.config }
+      let(:shared_credentials) do
+        Aws::SharedCredentials.new(path: Rails.root.join('spec', 'fixtures', 'aws_credentials'))
+      end
+
+      after do
+        Aws.config = {}
+      end
+
+      context 'profile us_west_2' do
+        let(:uno) do
+          {
+            'AWS_PROFILE' => 'us_west_2'
+          }
+        end
+
+        around do |example|
```

Comment:
Can move both these `around` blocks up one level and consolidate them.  This helps make it clear what each test is testing.

Or make all the `envs` names consistent and move ALL the `around` blocks to the top level.